src/main/scala/cromwell/engine/backend/jes/JesBackend.scala

```diff
@@ -104,30 +81,54 @@ class JesBackend extends Backend with LazyLogging {
     }
   }
 
+  sealed trait JesParameter {
+    def name: String
+    def gcs: String
+    def local: Path
+    def parameterType: String
+
+    final val isInput = this.isInstanceOf[JesInput]
```

Comment:
thought about it more, prob right thing here is a pattern match as I alluded to previously. more of a fp style polymorphism on adt vs oo style but guaranteed to work right due to sealed trait exhaustiveness checking